tracer/src/Datadog.Trace/Iast/EvidenceConverter.cs

```diff
@@ -59,29 +59,34 @@ private static string Substring(string value, Range range)
         return value.Substring(range.Start, range.Length);
     }
 
-    public override Evidence ReadJson(JsonReader reader, Type objectType, Evidence existingValue, bool hasExistingValue, JsonSerializer serializer)
+    public override Evidence? ReadJson(JsonReader reader, Type objectType, Evidence? existingValue, bool hasExistingValue, JsonSerializer serializer)
     {
         throw new NotImplementedException();
     }
 
-    public override void WriteJson(JsonWriter writer, Evidence evidence, JsonSerializer serializer)
+    public override void WriteJson(JsonWriter writer, Evidence? evidence, JsonSerializer serializer)
     {
+        if (evidence is null)
+        {
+            return;
+        }
+
         writer.WriteStartObject();
-        if (evidence.Ranges == null || evidence.Ranges.Length == 0)
+        if (evidence?.Ranges == null || evidence?.Ranges.Length == 0)
         {
             writer.WritePropertyName("value");
-            writer.WriteValue(evidence.Value);
+            writer.WriteValue(evidence?.Value);
         }
         else
         {
             writer.WritePropertyName("valueParts");
             if (_redactionEnabled)
             {
-                ToRedactedJson(writer, evidence.Value!, evidence.Ranges, evidence.Sensitive);
+                ToRedactedJson(writer, evidence?.Value!, evidence?.Ranges, evidence?.Sensitive);
             }
             else
             {
-                ToJsonTaintedValue(writer, evidence.Value!, evidence.Ranges);
+                ToJsonTaintedValue(writer, evidence?.Value!, evidence?.Ranges!);
```

Comment:
I think all these `?` should be `!` (at best). They actually _shouldn't_ be required at all as you've already checked for null at the start of the method 🤔

Reply:
Again, for some reason, the compiler shows an error even after checking for a null value

Reply:
Weird, we should use `!` though in those cases I think